helpers/uploadlogs.cpp

```diff
@@ -233,6 +236,153 @@ std::string errToText(err_t err)
     return "";
 }
 
+bool checkXpkiMtlsBasedLogUpload(){
+    if ( Utils::fileExists("/usr/bin/rdkssacli") &&
+            Utils::fileExists("/opt/certs/devicecert_1.pk12") ){
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+bool checkmTlsLogUploadFlag(){
+    bool ret=false;
+    RFC_ParamData_t param;
+    WDMP_STATUS wdmpStatus = getRFCParameter(const_cast<char *>("SystemServices"),TR181_MTLS_LOGUPLOAD, &param);
+    if (wdmpStatus == WDMP_SUCCESS || wdmpStatus == WDMP_ERR_DEFAULT_VALUE){
+        if( param.type == WDMP_BOOLEAN ){
+            if(strncasecmp(param.value,"true",4) == 0 ){
+                ret=true;
+            }
+        }
+    }
+    LOGINFO(" mTlsLogUpload.Enable = %s , call value %d ", (ret == true)?"true":"false", wdmpStatus);
+    return ret;
+}
+
+bool checkLogUploadBeforeDeepSleepFlag(){
+    bool ret=false;
+    RFC_ParamData_t param;
+    WDMP_STATUS wdmpStatus = getRFCParameter(const_cast<char *>("SystemServices"),TR181_LOGUPLOAD_BEF_DEEPSLEEP, &param);
+    if (wdmpStatus == WDMP_SUCCESS || wdmpStatus == WDMP_ERR_DEFAULT_VALUE){
+        if( param.type == WDMP_BOOLEAN ){
+            if(strncasecmp(param.value,"true",4) == 0 ){
+                ret=true;
+            }
+        }
+    }
+    LOGINFO(" mTlsLogUpload.Enable = %s , call value %d ", (ret == true)?"true":"false", wdmpStatus);
+    return ret;
+}
+
+
+bool getDCMconfigDetails(string &upload_protocol,string &httplink, string &uploadCheck){
+
+    string dcminfo;
+
+    if (!getFileContent(TMP_DCM_SETTINGS,dcminfo)) {
+        return false;
+    }
+    if (dcminfo.length() < 1){
+        return false;
+    }
+
+    smatch match;
+    string temp;
+    if (regex_search(dcminfo, match, regex("LogUploadSettings:UploadRepository:uploadProtocol=([^\\n]+)"))
+            &&  match.size() > 1) temp = trim(match[1]);
+    if (temp.size() > 0) upload_protocol = temp;
+
+    if (regex_search(dcminfo, match, regex("LogUploadSettings:UploadRepository:URL=([^\\n]+)"))
+            &&  match.size() > 1) temp = trim(match[1]);
+    if (temp.size() > 0) httplink = temp;
+
+    if (regex_search(dcminfo, match, regex("LogUploadSettings:UploadOnReboot=([^\\n]+)"))
+            &&  match.size() > 1) temp = trim(match[1]);
+    if (temp.size() > 0) uploadCheck = temp;
+
+    return true;
+}
+
+
+/* Call uploadSTBLogs.sh direclty by preparing the url
+ * and other details
+ */
```

Comment:
can't we reuse the uploadLogs() API that was added before. This should do the same as the script.

Reply:
We cant use the uploadLogs(),  as it is the basic version that uses the curl API,  not sure if MTLS itself is enabled or not,  we require event senders to provide notification to the maintenance manager in case if both log upload before deep sleep and maintenance manager log upload is simultaneously performing log upload. Also as per input from the ARCH team, they suggested to call the scripts right away from the setDevicePowerState.